src/main/java/uk/co/ramp/event/FormattedEventFactory.java

```diff
@@ -0,0 +1,101 @@
+package uk.co.ramp.event;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Service;
+import uk.co.ramp.event.types.*;
+import uk.co.ramp.people.Case;
+
+@Service
+public class FormattedEventFactory {
+
+    private static final Logger LOGGER = LogManager.getLogger(FormattedEventFactory.class);
+
+    public ImmutableFormattedEvent create(Event event) {
+
+        if (event instanceof VirusEvent) {
```

Comment:
The use of instanceof here seems like code smell. We may have to refactor but probably best to leave that for a separate PR as I think its not a simple fix

Reply:
Yeah I winced when I was writing it.